src/main/java/io/datanerds/avropatch/Patch.java

```diff
@@ -35,16 +45,29 @@ public boolean addOperation(Operation operation) {
         return Collections.unmodifiableList(operations);
     }
 
+    public Map<String, ?> getHeaders() {
+        return Collections.unmodifiableMap(headers);
+    }
+
     public byte[] toBytes() throws IOException {
         ByteArrayOutputStream outputStream = new ByteArrayOutputStream();
         Encoder binaryEncoder = EncoderFactory.get().directBinaryEncoder(outputStream, null);
-
-        writer.write(this, binaryEncoder);
+        SERIALIZER.writer.write(this, binaryEncoder);
         return outputStream.toByteArray();
     }
 
     public static Patch of(byte[] bytes) throws IOException {
-        DatumReader<Patch> reader = new ReflectDatumReader<>(SCHEMA);
-        return reader.read(null, DecoderFactory.get().binaryDecoder(bytes, null));
+        return SERIALIZER.reader.read(null, DecoderFactory.get().binaryDecoder(bytes, null));
```

Comment:
I assume we are not reusing the read "Patch" object here because it is being returned to an external method as it is, correct?